src/analytics/private/scan-metadata-workflow.ts

```diff
@@ -0,0 +1,274 @@
+          'appId.$': '$',
+        },
+      }),
+      outputPath: '$.Payload',
+    });
+
+    const checkScanMetadataStatusFn = this.checkScanMetadataStatusFn(props);
+    const checkScanMetadataStatusJob = new LambdaInvoke(this, `${this.node.id} - Check scan metadata job status`, {
+      lambdaFunction: checkScanMetadataStatusFn,
+      payload: TaskInput.fromObject({
+        'detail.$': '$.detail',
+      }),
+      outputPath: '$.Payload',
+    });
+
+    const waitX = new Wait(this, `${this.node.id} - Wait seconds`, {
+      /**
+         *  You can also implement with the path stored in the state like:
+         *  sfn.WaitTime.secondsPath('$.waitSeconds')
+      */
+      time: WaitTime.duration(Duration.seconds(30)),
+    });
+
+    const scanMetadataJobFailed = new Fail(this, `${this.node.id} - scan metadata job fails`, {
+      cause: 'ScanMetadata Job Failed',
+      error: 'DescribeJob returned FAILED',
+    });
+
+    const scanMetadataCompleted = new Succeed(this, `${this.node.id} - scan metadata job completes`);
+
+    const storeMetadataIntoDDBFn = this.storeMetadataIntoDDBFn(props);
+    const storeMetadataIntoDDBJob = new LambdaInvoke(this, `${this.node.id} - store scan metadata job status`, {
+      lambdaFunction: storeMetadataIntoDDBFn,
+      payload: TaskInput.fromObject({
+        'detail.$': '$.detail',
+      }),
+      outputPath: '$.Payload',
+    }).next(
+      new Choice(this, 'Check if store metadata completed')
+        .when(Condition.stringEquals('$.detail.status', 'SUCCEED'), scanMetadataCompleted)
+        .otherwise(scanMetadataJobFailed),
+    );
+
+    // Create sub chain
+    const subDefinition = scanMetadataJob
+      .next(waitX)
+      .next(checkScanMetadataStatusJob)
+      .next(
+        new Choice(this, `${this.node.id} - Check if scan metadata job completes`)
+          .when(Condition.stringEquals('$.detail.status', 'FAILED'), scanMetadataJobFailed)
+          .when(Condition.stringEquals('$.detail.status', 'ABORTED'), scanMetadataJobFailed)
+          .when(Condition.stringEquals('$.detail.status', 'FINISHED'), storeMetadataIntoDDBJob)
+          .when(Condition.stringEquals('$.detail.status', 'NO_JOBS'), scanMetadataCompleted)
+          .otherwise(waitX));
+
+    const doScanMetadataJob = new Map(
+      this,
+      `${this.node.id} - Do scanMetadata job`,
+      {
+        maxConcurrency: 1,
+        itemsPath: '$.appIdList',
+      },
+    );
+    doScanMetadataJob.iterator(subDefinition);
+
+    const doNothing = new Succeed(this, `${this.node.id} - Do Nothing`);
+    const checkJobExist = new Choice(this, `${this.node.id} - Check if job exists`)
+      .when(Condition.isNotPresent('$.appIdList'), doNothing)
+      .when(Condition.isPresent('$.appIdList'), doScanMetadataJob)
+      .otherwise(doNothing);
+
+    const definition = getJobList.next(checkJobExist);
+
+    // Create state machine
+    const scanMetadataStateMachine = new StateMachine(this, 'ScanMetadataStateMachine', {
```

Comment:
Is it possible adding to description for this workflow purpose and project information?

Reply:
Add the description